src/Features/Core/Portable/SplitOrMergeIfStatements/Consecutive/AbstractMergeConsecutiveIfStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SplitOrMergeIfStatements
+{
+    internal abstract class AbstractMergeConsecutiveIfStatementsCodeRefactoringProvider
+        : AbstractMergeIfStatementsCodeRefactoringProvider
+    {
+        // Converts:
+        //    if (a)
+        //        Console.WriteLine();
+        //    else if (b)
+        //        Console.WriteLine();
+        //
+        // To:
+        //    if (a || b)
+        //        Console.WriteLine();
+
+        // Converts:
+        //    if (a)
+        //        return;
+        //    if (b)
+        //        return;
+        //
+        // To:
+        //    if (a || b)
+        //        return;
```

Comment:
good comments.  potentially worth calling out that the body-statements need to be equivalent (in case someone doesn't realize that).

Reply:
➡️ Fixed in 632ac93e6bf247f143b8e88e21c8c6d03cf731e8